spec/services/debate_maker_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+describe DebateMaker do
+  describe "#call" do
+    let(:params) do
+      {
+        topic: "Our debate topic",
+        channel_name: "slack_debate_channel"
+      }
+    end
+
+    subject do
+      DebateMaker.call(params)
```

Comment:
I am thinking if it wouldn't be better to define subject as instance of `DebateMaker`. In one test you are relying on the behavior that when calling `subject` the block will be executed. Also now the evaluated subject will be an instance of `Debate` instead of `DebateMaker` which is the real subject of tests. I would probably write it this way:

```ruby
let(:debate) { subject.call params }
subject { described_class.new }
```

and then in tests:

```ruby
expect { subject.call(params) }.to change ...

# and

expect(debate.slug) ...
```